source/agora/utils/Test.d

```diff
@@ -624,62 +627,274 @@ unittest
+public struct OutputRef
+{
+    ///
+    public Hash hash () const nothrow @safe
+    {
+        return hashFull(this.tx);
+    }
 
-    Generate a new transaction that evenly splits the input accross parties
+    ///
+    public ref inout(Output) output () inout @safe pure nothrow @nogc
+    {
+        return this.tx.outputs[index];
+    }
 
-    The `input` transaction will be split evenly in `toward.length` outputs,
-    each of which will be controlled by a key in `toward`.
-    If the sum of outputs in the transaction is not a multiple of
-    `toward.length`, the leftover `Amount` will be added to
-    the output for `toward[0]`.
-    Like other testing utilities, if an error happens (e.g. `input` is invalid,
-    or `from` is missing some keys), an `assert` will be triggered.
+    /// The `Transaction` with the `Output` being referenced
+    public Transaction tx;
 
-    Params:
-        type = Type of transaction to generate (frozen or simple payment)
-        input = Transaction to spend completely
-        from = Array of keys controlling all the outputs in `input`
-        toward = Receivers for the newly-created output
+    /// The index of the `Output` being referenced
+    public uint        index;
+}
 
-    Returns:
-        A newly created, valid, and signed `Transaction` spending `input`.
+/// Get an `OutputRef` from a `Transaction`
+public inout(OutputRef) oref (inout Transaction tx, uint index)
+{
+    assert(tx.outputs.length > index);
+    return inout(OutputRef)(tx, index);
+}
+
+/// Get a range of all `OutputRef` from a `Transaction`
+public auto oref (const Transaction tx)
+{
+    import std.range;
+    return iota(tx.outputs.length).map!(idx => const(OutputRef)(tx, cast(uint)idx));
+}
+
+/*******************************************************************************
+
+    An helper utility to build transactions
+
+    This utility uses well-known keys (`WK.Keys`) for signature and expect
+    output created (via `split`, `merge`, etc...) to use them as addresses.
 
 *******************************************************************************/
 
-public Transaction split (TxType type = TxType.Payment)
-    (const ref Transaction input, scope const KeyPair[] from,
-     scope const PublicKey[] toward...)
-    @safe
+public struct TxBuilder
 {
-    Amount amount;
-    if (!input.getSumOutput(amount))
-        assert(0, "Invalid transaction passed to `split`");
+    /***************************************************************************
+
+        Signs the inputs.
+
+        Params:
+            type = type of `Transaction`
+            unknowns = unknown `KeyPair`s.
+                finds in `unknowns` first. If not, finds it in 'WK'.
+
+        Returns:
+            returns the final built `Transaction`.
+
+    ***************************************************************************/
+
+    public Transaction sign (TxType type = TxType.Payment,
+        scope const(KeyPair)[] unknowns = null) @safe
```

Comment:
Why is this needed ?

Reply:
The parameters unknowns are added for `system integration test` and
and `source/tests/unit/*.d`.

Reply:
They need a genesis KeyPair of coin-net.